core/lib/vm_interface/src/vm.rs

```diff
@@ -46,13 +48,13 @@ pub trait VmInterface {
 
     /// Execute batch till the end and return the result, with final execution state
     /// and bootloader memory.
-    fn finish_batch(&mut self) -> FinishedL1Batch;
+    fn finish_batch(&mut self, pubdata_builder: Option<Rc<dyn PubdataBuilder>>) -> FinishedL1Batch;
```

Comment:
I added `None` so caller doesn't have to create some pubdata builder if it calls VM that doesn't need it. But you are right, it shouldn't cause any issues for them, they can use e.g. `RollupPubdataBuilder::new(Address::zero())`. So I will remove optionality to avoid some boilerplate